sale_promotion_domain_product_discount/models/sale_order.py

```diff
@@ -0,0 +1,77 @@
+from odoo import _, models
+from odoo.tools.misc import formatLang
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_reward_values_discount(self, program):
+        """
+        Override core method to check and apply new discount apply rule
+        """
+        if program.discount_apply_on == "domain_product":
```

Comment:
please avoid such huge methods. Split the logic in smaller, more readable and more test-able methods.

Reply:
I'm trying to. However this is not clear for me how to do it in this case

Reply:
I have no idea how to refactor it. In any case, the original method from Odoo has more code than here.

Reply:
I posted some examples... then you can use your fantasy :wink: 

Is not a blocker but your future selves and everyone else future self will be happy about this in the long term.
Is going to be more readable and easier to add unit tests and/or debug.